observer/baseNodeProvider.go

```diff
@@ -61,81 +72,129 @@ func (bnp *baseNodeProvider) UpdateNodesBasedOnSyncState(nodesWithSyncStatus []*
 	bnp.mutNodes.Lock()
 	defer bnp.mutNodes.Unlock()
 
-	shardIDs := getSortedSliceIDsSlice(bnp.nodesMap)
-	syncedNodes, outOfSyncNodes, err := computeSyncedAndOutOfSyncNodes(nodesWithSyncStatus, shardIDs)
+	syncedNodes, syncedFallbackNodes, outOfSyncNodes, err := computeSyncedAndOutOfSyncNodes(nodesWithSyncStatus, bnp.shardIds)
```

Comment:
update the comment of `UpdateNodesBasedOnSyncState` by explaining all the changes that could happen (regular <--> fallback)